tools/scripts/dpm.py

```diff
@@ -0,0 +1,176 @@
+cert_opt   = ca_default    # Certificate field options
+
+# Extension copying option: use with caution.
+# copy_extensions = copy
+
+# Extensions to add to a CRL. Note: Netscape communicator chokes on V2 CRLs
+# so this is commented out by default to leave a V1 CRL.
+# crlnumber must also be commented out to leave a V1 CRL.
+# crl_extensions  = crl_ext
+
+default_days  = 365      # how long to certify for
+default_crl_days= 30      # how long before next CRL
+default_md  = sha256    # which md to use.
+preserve  = no      # keep passed DN ordering
+
+# A few difference way of specifying how similar the request should look
+# For type CA, the listed attributes must be the same, and the optional
+# and supplied fields are just that :-)
+policy    = policy_match
+
+# For the CA policy
+[ policy_match ]
+emailAddress    = optional
+commonName    = supplied
+organizationalUnitName  = optional
+organizationName  = optional
+stateOrProvinceName  = optional
+countryName    = optional
+
+# For the 'anything' policy
+# At this point in time, you must list all acceptable 'object'
+# types.
+[ policy_anything ]
+emailAddress    = optional
+commonName    = supplied
+organizationalUnitName  = optional
+organizationName  = optional
+localityName    = optional
+stateOrProvinceName  = optional
+countryName    = optional
+
+[ req ]
+prompt                  = no
+#default_bits    = 1024
+#default_keyfile   = privkey.pem
+distinguished_name  = req_distinguished_name
+#attributes    = req_attributes
+req_extensions  = v3_ca  # The extentions to add to the self signed cert
+
+[ v3_ca ]
+basicConstraints        = critical, CA:TRUE
+subjectKeyIdentifier    = hash
+#authorityKeyIdentifier  = keyid:always, issuer:always
+keyUsage                = critical, digitalSignature, keyCertSign
+
+[ req_distinguished_name ]
+countryName         = US
+stateOrProvinceName = MO
+localityName        = Saint Louis
+0.organizationName  = Object Computing ({self.name} CA)
+commonName          = Object Computing ({self.name} CA)
+emailAddress        = info@objectcomputing.com
+'''
+        with open(self.config_path, 'w') as file:
+            print(config_content, file=file)
+        index_path = self.path / 'index.txt'
+        index_path.touch()
+        serial_path = self.path / 'serial'
+        serial_path.write_text('01')
+        subprocess.check_call(['openssl', 'ecparam', '-name', 'prime256v1', '-genkey', '-out', self.key_path])
+        subprocess.check_call(['openssl', 'req', '-config', self.config_path, '-new', '-key', self.key_path, '-out', csr_path])
+        subprocess.check_call(['openssl', 'x509', '-req', '-days', '3650', '-copy_extensions=copyall', '-in', csr_path, '-signkey', self.key_path, '-out', self.cert_path])
+
+    def gencert(self, name, sn):
+        identity_path = dpm_path / 'identity'
+        path = identity_path / name
+        if os.path.isdir(path):
+            print(f'{path} already exists')
+            return
+        os.makedirs(path)
```

Comment:
```suggestion
        path.mkdir(parents=True)
```

It will raise if the directory exists.